go/vt/vtgate/planbuilder/operators/route.go

```diff
@@ -742,7 +742,14 @@ func (r *Route) FindCol(ctx *plancontext.PlanningContext, expr sqlparser.Expr, _
 }
 
 func (r *Route) GetColumns(ctx *plancontext.PlanningContext) []*sqlparser.AliasedExpr {
-	return truncate(r, r.Source.GetColumns(ctx))
```

Comment:
Why is this changed? The `truncate` call here already would implement the logic? You also replaced it in other places? Are you sure that is correct and the fix is not to be implemented somewhere else? Similarly, since you added a non-working test, I'm very unsure that the bug here is fully understood and that this fix is right?

Reply:
Yeah, just validate but the changes here make no sense. They do not address the bug and are a no op essentially since we already have this logic.

The real problem is that on the outer aggregate, the planner does to set the proper `ResultColumns` at all.